specreduce/extract.py

```diff
@@ -13,6 +13,66 @@
 __all__ = ['BoxcarExtract']
 
 
+def _get_boxcar_weights(center, hwidth, npix):
```

Comment:
if these methods are going to be used outside of this file, they should be renamed to remove the leading `_` and added to `__all__`.

Reply:
these are methods we need in both extract and background, but probably shouldn't be user-facing.  Do you still want me to do this and we just won't "advertise" them in the docs?  Or is there some better way to handle this type of situation?

Reply:
if they are going to only used within `specreduce`, then they're fine as-is. my personal style is to keep things private within files, but the official python guidelines treat privacy at the module level. the leading `_` and exclusion from `__all__` should be sufficient to keep them from being advertised.